pkg/v1/remote/image.go

```diff
@@ -107,7 +109,19 @@ func (r *remoteImage) RawConfigFile() ([]byte, error) {
 	return r.config, nil
 }
 
+func (r *remoteImage) Descriptor() (*v1.Descriptor, error) {
+	if r.descriptor != nil {
+		return r.descriptor, nil
+	}
+
+	// NOTE(jonjohnsonjr): We should never get here because the public entrypoints
```

Comment:
Why not change the tests that directly deal with remoteImage?

Reply:
I'm kinda torn on this because my comment just so happens to be true, and there's nothing in the type system to enforce this invariant :/

Reply:
Alright I came up with something gross but slightly better